plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/thrift/ThriftMetastoreClientFactory.java

```diff
@@ -20,6 +20,6 @@
 
 public interface ThriftMetastoreClientFactory
 {
-    ThriftMetastoreClient create(URI uri, Optional<String> delegationToken)
+    ThriftMetastoreClient create(URI uri, Optional<String> delegationToken, String catalogName)
```

Comment:
instead you could just change ClientFactory constructors to inject StaticMetastoreConfig, right?

Reply:
`StaticMetastoreConfig` will be used only for the `catalogName`, as for `metastoreUri` config has `metastoreUris` and client is created for each uri, therefore can't use the config in the `ThriftMetastoreClientFactory .create`

```
    public ThriftMetastoreClient createMetastoreClientFor(Optional<ConnectorIdentity> identity)
            throws TException
    {
        TException lastException = null;
        for (URI metastoreUri : metastoreUris) {
            try {
                return clientProvider.create(metastoreUri, Optional.empty(), catalogName);
            }
            catch (TException e) {
                lastException = e;
            }
        }
        throw lastException;
    }
```